Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialTypeAsset.cpp

```diff
@@ -76,6 +76,34 @@ namespace AZ
             AssetInitBus::Handler::BusDisconnect();
         }
 
+        bool MaterialTypeAsset::InitializeNonSerializedData()
+        {
+            if (m_isNonSerializedDataInitialized)
+            {
+                return true;
+            }
+            for (auto& shaderItem : m_generalShaderCollection)
+            {
+                if (!shaderItem.InitializeShaderOptionGroup())
+                {
+                    return false;
+                }
+            }
+
+            for (auto& materialPipelinePair : m_materialPipelinePayloads)
+            {
+                for (auto& shaderItem : materialPipelinePair.second.m_shaderCollection)
+                {
+                    if (!shaderItem.InitializeShaderOptionGroup())
+                    {
+                        return false;
+                    }
+                }
+            }
```

Comment:
Can these loops be moved into a function on the shader collection class? 

ShaderCollection::InitializeShaderOptionGroup

Reply:
Done.